libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/LegacyMapboxNavigationInstanceHolder.kt

```diff
@@ -0,0 +1,46 @@
+package com.mapbox.navigation.core.internal
+
+import androidx.annotation.UiThread
+import androidx.annotation.VisibleForTesting
+import com.mapbox.navigation.core.MapboxNavigation
+import java.util.concurrent.CopyOnWriteArraySet
+
+fun interface MapboxNavigationCreateObserver {
+
+    fun onCreated(mapboxNavigation: MapboxNavigation)
+}
+
+@Deprecated("Used to keep track of MapboxNavigation instances created via deprecated methods")
+@UiThread
+object LegacyMapboxNavigationInstanceHolder {
```

Comment:
I don't think we should add this object. 

I understand that `MapboxNavigationApp` may not be perfect. But it was also made because `MapboxNavigation` is not perfect. Introducing another imperfect object will just make the problem worse.

Reply:
There is nothing this object is doing, that cannot be done by `MapboxNavigationApp` and `MapboxNavigationObserver`. Considering that it is not adding anything and just adding more opinions, we should not add it

Reply:
It can't. There are ways of creating `MapboxNavigation` that `MapboxNavigationApp` doesn't track. Like using the constructor directly. Or do you suggest that `MapboxNavigationApp` should track this as well?

Reply:
I'm suggesting to not into a constructor callback. MapboxAudioGuidance already does that, and it can be used with or without MapboxNavigationApp.

If a developer wants to use predownloaded instructions they would need to manually fetch them after MapboxNavigation has been created. I'm saying there is no need to add a new automatic link for the constructor

Reply:
It will not be needed if we do what Vadzim suggested: only provide out-of-the-box predownloading for high level API (MapboxAudioGuidance) but for low-level API users (that use `MapboxSpeechAPI` directly) just add a new method. In the current implementation it's not the case. But I'm working on it now.